src/components/Location/LocationSearch.tsx

```diff
@@ -50,14 +57,31 @@ export function LocationSearch({
     <Popover open={open} onOpenChange={setOpen}>
       <PopoverTrigger asChild disabled={disabled}>
         <div
-          className="w-full h-9 px-3 rounded-md border text-sm flex items-center justify-between cursor-pointer"
+          className={cn(
+            "w-full h-9 px-3 rounded-lg border border-gray-300 text-sm shadow-sm transition-colors bg-white flex items-center justify-between cursor-pointer text-gray-950",
+            className,
+          )}
           role="combobox"
           aria-expanded={open}
         >
-          {stringifyNestedObject(value || { name: "" }) || "Select location..."}
+          {stringifyNestedObject(value || { name: "" }) || (
+            <span className="text-gray-500">Select location...</span>
+          )}
+          {value && (
+            <Button
+              onClick={(e) => {
+                e.stopPropagation();
+                onSelect(null);
+              }}
+              variant="ghost"
+              className="hover:bg-transparent"
+            >
+              <CareIcon icon="l-times" className="h-4 w-4" />
+            </Button>
+          )}
         </div>
       </PopoverTrigger>
-      <PopoverContent className="w-[400px] p-0">
+      <PopoverContent className="min-w-[250px] max-w-[400px] w-full p-0">
```

Comment:
check `--radix-popover-trigger-width` in Shadcn popover documentation. You can also check codebase for other examples.